alternator/executor.cc

```diff
@@ -1127,7 +1127,7 @@ future<executor::request_return_type> executor::update_table(client_state& clien
 
     for (auto& s : unsupported) {
         if (rjson::find(request, s)) {
-            co_return coroutine::make_exception(api_error::validation(s + " not supported"));
+            co_await coroutine::return_exception(api_error::validation(s + " not supported"));
```

Comment:
@bhalevy 
I'm really confused now. Why is co_await return_exception better in any sense than co_return make_exception?
Isn't the co_return's intentions clearer?

Reply:
`co_return make_exception` is clearer, but it doesn't work for `future<>` type (without any template parameters) due to technicalities in coroutine specification, while `co_await return_exception` works everywhere - I think it's ultimately better to use a single consistent idiom

Reply:
== The reason we deprecated coroutine::make_exception was to keep just one interface that satisfies all use cases, rather than having 2 that are confusing (especially with regards to returning an exception_ptr where Avi wanted to have a sepcial entry point for it: `co_await coroutine::return_exception_ptr(std::exception_ptr)`, while `co_return coroutine::make_exception_ptr(std::exception_ptr)` makes no sense.

For clarity, there's still the ability to `co_return coroutine::exception(std::exception_ptr)`.
I used it in this patch only where it is required (since we can't co_await in a catch clause),
and often it required explicitly calling `std::make_exception_ptr` on the given exception object.